packages/app/src/app/pages/Dashboard/Components/Selection/ContextMenus/SandboxMenu.tsx

```diff
@@ -43,10 +43,21 @@ export const SandboxMenu: React.FC<SandboxMenuProps> = ({
 
   // TODO(@CompuIves): remove the `item.sandbox.teamId === null` check, once the server is not
   // responding with teamId == null for personal templates anymore.
-  const hasAccess = React.useMemo(
-    () => item.sandbox.teamId === activeTeam || item.sandbox.teamId === null,
-    [item, activeTeam]
-  );
+  const hasAccess = React.useMemo(() => {
+    if (item.sandbox.teamId === activeTeam) {
+      return true;
+    }
+
+    if (item.sandbox.teamId === null) {
+      if (!item.sandbox.authorId) {
+        return false;
+      }
+
+      return true;
```

Comment:
We might need to check if author id on the sandbox is the same as current user ID. Other than that this looks good!

Reply:
But in teams you can delete other people's templates, no?

Reply:
Gooooood point... LGTM!